---
visualizer_frame: $(arg visualizer_frame)
config:
  layer_z_step: 5.5  # unit separation between layers
  mesh_edge_break_ratio: 0.5
  mesh_layer_offset: 0.0
  collapse_layers: false
  places_colormap_min_distance: 0.0
  places_colormap_max_distance: 3.0
  places_colormap:
    min_hue: 0.0
    max_hue: 0.5
    min_saturation: 0.9
    max_saturation: 0.8
    min_luminance: 0.6
    max_luminance: 0.85
  place_tasks:
    min_hue: 0.0
    max_hue: 0.13
    min_saturation: 0.7
    max_saturation: 0.9
    min_luminance: 0.5
    max_luminance: 0.8
  dynamic_layers:
    '2':
      visualize: false
  layer2:
    visualize: false
  layer3:
    # general
    z_offset_scale: 0.0
    visualize: false
    # nodes
    marker_scale: 0.3
    marker_color_mode: 1  # parent
    marker_alpha: 1.0
    use_sphere_marker: true
    use_label: true
    use_collapsed_label: false
    label_height: 1.0
    label_scale: 0.5
    use_bounding_box: false
    bounding_box_alpha: 0.5
    collapse_bounding_box: false
    # edges
    use_edge_source: false
    interlayer_edge_scale: 0.08
    interlayer_edge_alpha: 0.4
    interlayer_edge_use_color: true
    interlayer_edge_insertion_skip: 0
    intralayer_edge_scale: 0.005
    intralayer_edge_alpha: 0.5
    intralayer_edge_use_color: false
    intralayer_edge_insertion_skip: 0
  layer4:
    # general
    z_offset_scale: 4.2
    visualize: false
    # nodes
    marker_scale: 0.2
    marker_color_mode: 0  # default
    marker_alpha: 0.8
    use_sphere_marker: false
    use_label: false
    use_collapsed_label: false
    label_height: 1.25
    label_scale: 0.3
    use_bounding_box: false
    bounding_box_alpha: 0.5
    collapse_bounding_box: false
    # edges
    use_edge_source: true
    interlayer_edge_scale: 0.08
    interlayer_edge_alpha: 0.4
    interlayer_edge_use_color: true
    interlayer_edge_insertion_skip: 0
    intralayer_edge_scale: 0.01
    intralayer_edge_alpha: 0.5
    intralayer_edge_use_color: false
    intralayer_edge_insertion_skip: 0
  layer5:
    visualize: false
llm_plugins:
  dsg_mesh: MeshPlugin
  llm_places: LLMPlacesVisualizer
dsg_mesh:
  label_colormap: ''
  color_by_label: false
llm_places:
  min_score: 0.20
  max_score: 0.35
  use_fixed_range: true
  metric:
    type: cosine  # lerf
    cannonical_features:
      type: RosEmbeddingGroup
      service_name: /get_cannonical

